Merge Stroke-Prediction into Main

lawrenceadams opened the pull request.

chris-lovejoy commented:
Hey Laurence,

I've worked through this and made a few edits. The main change I made was breaking up the big block of text in Part 3 about Feature Encoding, and integrate some code (even if basic) amongst it.

I've put a few "TODO" comments - mostly minor tweaks that I would recommend, or a bug that I found (because I think I'm using a diff version of pandas to you).

One point I didn't mention in the exercise is that the images in the Data Encoding section aren't rendering for me - are these loading images from your local computer that aren't pushed to github?

The other main wider point is to add more (and make more explicit) the points where people need to write their own code. I saw for example where they should define the label_columns. But I think that the required answer may already be there? I'd propose we make a note of the answers outside of the repo, and then set these up for student completion.

Sorry for the long block of text! Let me know what you think of this, and if happy to work through the TODOs (both in text above and in the notebook itself). Once those are done, we'll merge this onto the main branch.

chris-lovejoy commented:
When I'm loading the notebook directly, I'm getting this error too - which presumably relates to one of the points I flagged, about the image not being found (and whether it needs to be in the repo?)

<img width="1264" alt="Screenshot 2023-03-02 at 22 19 51" src="https://user-images.githubusercontent.com/35599865/222572394-0dee3677-8681-49cc-afba-a4c1d7de0b5b.png">

lawrenceadams commented:
Thank you for all the input @chris-lovejoy! Working on some of it now

Todo:

- [X] Fix Images 
    - Fixed in b0c6dcf
- [ ] Add more exercise bits
- [x] Update keyword arguments for df.corr() ? Drop entirely
    - Fixed in 2027fb112a3eeb83e4e7549347c4184b28a2239d
- [x] Resolve error messages
    - Dependent on #12 
- [ ] Decide what models students should employ themselves
- #12 
- #7
